alumet/tests/late_source_creation.rs

```diff
@@ -0,0 +1,123 @@
+
+use alumet::{agent::{AgentBuilder, AgentConfig}, measurement::{MeasurementAccumulator, MeasurementPoint, Timestamp}, metrics::TypedMetricId, pipeline::{elements::error::PollError, trigger::TriggerSpec}, plugin::{rust::{deserialize_config, serialize_config, AlumetPlugin}, AlumetPluginStart, AlumetPostStart, ConfigTable}, resources::{Resource, ResourceConsumer}, static_plugins, units::{PrefixedUnit, Unit}};
+use anyhow::Context;
+use serde::{Deserialize, Serialize};
+use std::{thread, time::{self, Duration}};
+
+pub struct Metrics {
+    pub time_tot: TypedMetricId<u64>,
+}
+
+pub struct MyTestPluginLateMetricCreation {
+    config: Config,
+    metrics: Option<Metrics>,
+}
+
+#[derive(Debug)]
+struct MyTestSourcePlugin {
+    value: TypedMetricId<u64>,
+}
+
+#[derive(Deserialize, Serialize)]
+struct Config {
+    /// Initial interval between two cgroup measurements.
+    #[serde(with = "humantime_serde")]
+    poll_interval: Duration,
+}
+
+impl AlumetPlugin for MyTestPluginLateMetricCreation {
+    fn name() -> &'static str {
+        "late_source_creation"
+    }
+
+    fn version() -> &'static str {
+        env!("CARGO_PKG_VERSION")
+    }
+
+    fn default_config() -> anyhow::Result<Option<ConfigTable>> {
```

Comment:
IMO the test doesn't even need a configuration. You can arbitrary values :)

Reply:
it was to be as close as possible to a real plugin